### ISSUE TITLE
unwind

### ISSUE THREAD

**Issue by ndowens** [benfred/py-spy#334]
Seems libunwind is missing as a dependency? In voidlinux, I had to add libunwind-devel to makedepends in order to build successfully since it seems py-spy links against it "-lunwind"

**Comment by benfred** [comment]
Yeah - we need libunwind on x86_64 linux right now (and hopefully also on arm processors .

I'm assuming for voidlinux that you want to use the libunwind packaged with the OS? How would you expect this to be specified here?

I'm distributing a statically linked version using musl where I build libunwind myself on pypi. I could add this to the build.rs build script to make it easier for others, but I'm guessing this won't help you that much.

**Comment by ndowens** [comment]
How would it be added in build.rs so I can test that

**Comment by benfred** [comment]
Are you building the statically linked version with musl (```cargo build --target x86_64-unknown-linux-musl --release```)  - or the dynamically linked version with glibc?

**Comment by ndowens** [comment]
Looks like void does: build --release --target ${RUST_TARGET} ${configure_args} where RUST_TARGET I'd imagine is the arch

**Comment by BwL1289** [comment]
@benfred in my case, this issue stems from https://github.com/benfred/remoteprocess looking for the following at link time when the `unwind` feature is enabled (which py-spy enables by default on Linux):
```bash
unwind-ptrace
unwind-aarch64
```

If you're using LLVM libunwind instead of GNU libunwind, it does not split into multiple shared libraries. It only ships `libunwind.so`. GNU libunwind provides:
```bash
libunwind.so
libunwind-aarch64.so
libunwind-ptrace.so
```

So the linker fails to find the split components. 

Worth noting: even if you work around the link error by symlinking LLVM's libunwind.so under those names, LLVM libunwind is typically built without `ptrace` support (`_UPT_*` symbols absent), so `--native` will silently not work at runtime anyway. A cleaner fix would be building with `--no-default-features` to disable the unwind feature entirely when GNU libunwind is not available.

Related:
- https://github.com/benfred/py-spy/issues/571
- https://github.com/benfred/py-spy/issues/86
